### ISSUE TITLE
Trouble decrypting wallet

### ISSUE THREAD

**Issue by AsaBarthMaron** [langerhans/dogecoin-wallet-new#87]
I have previously had success with this method (https://github.com/langerhans/dogecoin-wallet-new/issues/68#issuecomment-769800820) for decrypting a wallet backup file. However I am now having no luck with a new wallet file. 

It is possible I am using the wrong password, but I am getting slightly different error messages when I try any other password, and I think the one I have in mind is in fact correct.

Upon running:
`openssl enc -d -aes-256-cbc -md md5 -a -in YOUR-dogecoin-wallet-backup-FILENAME -out dogecoin-wallet-decrypted`

With what I think is the correct password, I get:
`bad decrypt
4492775020:error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt:evp_enc.c:536:`

But when I use literally any other password, I get:
`bad decrypt
4591435372:error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt:evp_enc.c:531:`

So the first numbers (4492775020, 4591435372) are always different but the last number is always 536 for the 'correct' password but '531' for the 'incorrect' one.

This is on a machine with OpenSSL 1.0.2o 

I also tried on a different machine with OpenSSL 1.0.2h and got similar but slightly different error messages

'Correct' password:
`bad decrypt
4294956672:error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt:evp_enc.c:534`

Any other password:
`bad decrypt
4294956672:error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt:evp_enc.c:529:`

Again, it's always 534 for the 'correct' one and '529' for any other. 

So I am left wondering/unsure if this is an issue with the decryption or password, or worse if the file is corrupted. Thanks for the help!

**Comment by sandman2448** [comment]
I'm using the android version and I have a similar situation.

My back up is from 2014.

When I use a random password I get error wallet could not be restored: Could not decrypt bytes. Bad Password?
When I use what I think is the correct password I get "Protocol message contained an invalid tag (zero). Bad Password?

Do I need to use an older version of the wallet or something?

**Comment by btc2dogecom** [comment]
> I'm using the android version and I have a similar situation.
> 
> My back up is from 2014.
> 
> When I use a random password I get error wallet could not be restored: Could not decrypt bytes. Bad Password?
> When I use what I think is the correct password I get "Protocol message contained an invalid tag (zero). Bad Password?
> 
> Do I need to use an older version of the wallet or something?

how much do you have in this wallet backup? the backup data has problem. it is not caused by the password. it is the backup data itself. can't decrypt it directly with the app or openssl . but there is still chance to recover. contact us via: www.btc2doge.com if your balance worth the time. if small, i would advise you buy some doge instead.

**Comment by AsaBarthMaron** [comment]
> I'm using the android version and I have a similar situation.
> 
> My back up is from 2014.
> 
> When I use a random password I get error wallet could not be restored: Could not decrypt bytes. Bad Password?
> When I use what I think is the correct password I get "Protocol message contained an invalid tag (zero). Bad Password?
> 
> Do I need to use an older version of the wallet or something?

This is a little different from my current problem, but I have had this exact problem in the past and have been able to decrypt it with openssl. I am happy to help walk you through that. You can find my email on my github profile.

**Comment by sandman2448** [comment]
I'd really appreciate some help. However, if you don't mind it might be better to comment here so others with the same problem can see?

**Comment by gilgongo** [comment]
@AsaBarthMaron Could you post your instructions publicly? I may also have the same issue (backup from 2014 not being able to restore).

**Comment by AsaBarthMaron** [comment]
@sandman2448 @gilgongo Yes, apologies for the delay!

So first off you just need to have some method of using openssl. If you're on mac or linux it is just a matter of opening the command line. If you're on windows there's a few different ways to get openssl running but the easiest way I found is just to download the [git bash](https://git-scm.com/downloads).

Once you have that you can go about decrypting the wallet backup file using the command posted [here](https://github.com/langerhans/dogecoin-wallet-new/issues/68#issuecomment-769800820):
`openssl enc -d -aes-256-cbc -md md5 -a -in YOUR-dogecoin-wallet-backup-FILENAME -out dogecoin-wallet-decrypted`
Though importantly if you're using git bash on windows you will need to append `winpty` to the beginning:
`winpty openssl enc -d -aes-256-cbc -md md5 -a -in YOUR-dogecoin-wallet-backup-FILENAME -out dogecoin-wallet-decrypted`
It will then ask you for your encryption password, and save the decrypted file to `dogecoin-wallet-decrypted`

`dogecoin-wallet-decrypted` Should look something like:
```
# KEEP YOUR PRIVATE KEYS SAFE! Anyone who can read this can spend your Bitcoins. 
52-CHARACTER-LONG-ALPHANUMERIC 2014-01-20T21:32:56Z
````
The 52 character long string should start with a 'Q'. If you get to this stage you have completed the hard part and will definitely be able to get your doge!

There are a few methods to import a private key directly, but the one I like is using https://my.dogechain.info/#/overview. There are two methods here. Firstly you can create (or log into an existing) wallet (don't lose the wallet ID and password though haha). Then you can go to 'Import / Export',  enter the private key and press 'Import private key'. That should be it! You will then have all your doge in this new wallet. From there you can send it to a new wallet if you'd like with the 'send' option. The second method is to skip the dogechain wallet step and select the 'Redeem paper wallet' option on https://my.dogechain.info/#/overview. Here you will enter the private key directly, and it will then ask you to what public wallet address you want to send the contents of that private key. This will require you to already have another wallet set up elsewhere. I've used both methods successfully. 

I also have a reddit post where I explained some of this, albeit probably less thoroughly than I just did here: https://www.reddit.com/r/dogecoin/comments/lkaexc/dogecoin_daily_discussion_feb_15th/gnllm7o/?context=3

Donations: DRJegrY8y1Bw2AUd8w7hZErSCrX7SU5fLK

**Comment by gilgongo** [comment]
Nice! I managed to get my private key! Before I use it to get the coins though, I have two files. One that I've decrypted using your instructions - that was called "Dogecoin Wallet private key backup" (240 bytes). The other is a larger file called "Dogecoin Wallet backup" (32.2Kb) containing 420 lines of text each 77 characters long, apart from the last line ending in "==" which is shorter.

What's that?

**Comment by AsaBarthMaron** [comment]
@gilgongo Can you give me a little more info? What does the first line of the 240 byte file say? And what letter does the second line start with? (I am assuming there are two lines, as there should be). I believe that file should be the file with your private key, which you should be able to import on dogechain. I am not sure what the second file is, or how you ended up with two files. Did you start with more than one encrypted file? Or did decrypting somehow produce two? (shouldn't be possible)

**Comment by gilgongo** [comment]
The 240 byte file is the private key, and decrypted OK with the passphrase I had. I'm just curious what the second file 32.2Kb file is. Maybe it's some transaction data or something? I seem to remember I made the backbup from my phone, so it may be something the app was using. But I guess it does't matter as I've got my private key now :-)

**Comment by AsaBarthMaron** [comment]
@gilgongo Great, do you know how to go about recovering the doge from that key then?

**Comment by gilgongo** [comment]
Yes, as per your instructions, thanks! I tipped you some doge to that address in fact.

**Comment by louie333** [comment]
hi, I am quite a noob in these area, so can someone help me with this...

what do I have exactly to fill in that following spot in the command: in YOUR-dogecoin-wallet-backup-FILENAME ?

**Comment by gilgongo** [comment]
@louie333 What do you mean "that following spot" exactly?

**Comment by louie333** [comment]
What do I have to fill in myself overhere as a command exactly.' 
YOUR-dogecoin-wallet-backup-FILENAME'

**Comment by gilgongo** [comment]
That will be the name of the file that is the wallet you want to decrypt. For example, I had a file called "doge-wallet-bkp.dat" but yours might be called something else. In other words its simply the name of the file you have.

**Comment by louie333** [comment]
Ok thx, I dont have the .dat file thats the problem

I have the encrypted key and the password but don't know how to decrypt it...you have an idea?

**Comment by gilgongo** [comment]
OK sure. You just need to follow the instructions given here then: https://github.com/langerhans/dogecoin-wallet-new/issues/87#issuecomment-824173384

**Comment by louie333** [comment]
Thats the above comment where you need the .dat file for?

**Comment by gilgongo** [comment]
Yes, so you need to run than openssl command on your wallet file, basically. So if for example your wallet file is called "doge-backup.dat" then you'd run:

`openssl enc -d -aes-256-cbc -md md5 -a -in doge-backup.dat -out dogecoin-wallet-decrypted`

**Comment by louie333** [comment]
Maybe I was not clear in my previous comment...I don't have the .dat file anymore only the encrypted key with pass

**Comment by gilgongo** [comment]
Well, the key kinda *is* the wallet really, so you should be OK. Just try that command on the file you have and see if you get the decrypted plain text file with the 52 character key in it which you can then import it into a wallet (as per those instructions above).

**Comment by chmol** [comment]
Bumping that thread up because it allowed me to decrypt my 2014 key. 

Thanks a lot  @AsaBarthMaron

**Comment by hastiAu** [comment]
@AsaBarthMaron 
please help me

 I have received an email from Dogechain.info to transfer fund last week , when I want to login , when I fill my password , for just one time it send me 2FA code to my email , but it was just one time , the it does not send again , then I see a form that I send my name and city but then I could not login finally

then I wanted to try with 2FA.com application, I had back up file and I restore it to get code, but when I want to login to dogechain.info it does not accept the code that I receive it from 2FA application , and said it is invalid code , and it does not send 2fas code to my email too and I can not login to my account at all.

more over i have a exported encrypted file that I don't know how can I use it :the file start with :


salt(+xTuGfX+WWf/Tzb5AYv1TA==) ......

I send email to block.io but no body answer me.

just in case I create new wallet in dogechain but in that one I receive the code in my new email for new wallet

**Comment by doge2021** [comment]
> please help me
> 
> I have received an email from Dogechain.info to transfer fund last week , when I want to login , when I fill my password , for just one time it send me 2FA code to my email , but it was just one time , the it does not send again , then I see a form that I send my name and city but then I could not login finally
> 
> then I wanted to try with 2FA.com application, I had back up file and I restore it to get code, but when I want to login to dogechain.info it does not accept the code that I receive it from 2FA application , and said it is invalid code , and it does not send 2fas code to my email too and I can not login to my account at all.
> 
> more over i have a exported encrypted file that I don't know how can I use it :the file start with :
> 
> salt(+xTuGfX+WWf/Tzb5AYv1TA==) ......
> 
> I send email to block.io but no body answer me.
> 
> just in case I create new wallet in dogechain but in that one I receive the code in my new email for new wallet

@hastiAu Dogechain.info changed its platform and encryption method again. you can reach Stephen from [btc2doge.com](www.btc2doge.com). He helped me before. good luck.

**Comment by hastiAu** [comment]
> > please help me
> > I have received an email from Dogechain.info to transfer fund last week , when I want to login , when I fill my password , for just one time it send me 2FA code to my email , but it was just one time , the it does not send again , then I see a form that I send my name and city but then I could not login finally
> > then I wanted to try with 2FA.com application, I had back up file and I restore it to get code, but when I want to login to dogechain.info it does not accept the code that I receive it from 2FA application , and said it is invalid code , and it does not send 2fas code to my email too and I can not login to my account at all.
> > more over i have a exported encrypted file that I don't know how can I use it :the file start with :
> > salt(+xTuGfX+WWf/Tzb5AYv1TA==) ......
> > I send email to block.io but no body answer me.
> > just in case I create new wallet in dogechain but in that one I receive the code in my new email for new wallet
> 
> @hastiAu Dogechain.info changed its platform and encryption method again. you can reach Stephen from [www.btc2doge.com](http://www.btc2doge.com) He helped me before. good luck.

how can I trust him? he needs my personal information to recover my dogecoin

**Comment by hastiAu** [comment]
@doge2021 
how can I trust him? he needs my personal information to recover my dogecoin

**Comment by doge2021** [comment]
> @doge2021 how can I trust him? he needs my personal information to recover my dogecoin

@hastiAu it depends. some case need the wallet info, while other problems may not need the wallet info. in the past, for 2fa issue, i only need signed a form and he resolved my problem. but now, seems dogechain.info changed its platform and old method not work any more. i think your case is different. you may need provide them the file you mentioned start with: **salt(+xTuGfX+WWf/Tzb5AYv1TA==) ..**....
this maybe the last hope as i know many guys get their dogecoin locked by dogechain.info due to this issue. good luck

**Comment by hastiAu** [comment]
@AsaBarthMaron 
 
  
 my backup is some text that I have saved before on a text file , because I have created a wallet just on the Dogechain.info website
 
 and after this code I see this error:

`openssl enc -aes-256-cbc -base64 -pass pass:149149ssS -d -p -in ~/c/Users/Hasti/Desktop/key.txt -out enc.txt Can't open "C:/Users/Hasti/c/Users/Hasti/Desktop/key.txt" for reading, No such file or directory 403E0000:error:80000003:system library:BIO_new_file:No such process:../openssl-3.1.4/crypto/bio/bss_file.c:67:calling fopen(C:/Users/Hasti/c/Users/Hasti/Desktop/key.txt, r) 403E0000:error:10000080:BIO routines:BIO_new_file:no such file:../openssl-3.1.4/crypto/bio/bss_file.c:75:`

**Comment by gilgongo** [comment]
Are you using Cygwin? There's some path problem. You're giving it  `~/c/Users/Hasti/Desktop/key.txt`  but openssl is getting `"C:/Users/Hasti/c/Users/Hasti/Desktop/key.txt"` so it can't find that. 

Maybe try `in ~/Users/Hasti/Desktop/key.txt ` or just cd to `Desktop` and use `-in ./key.txt`

**Comment by hastiAu** [comment]
@gilgongo no I use Git Bash. but none of these path does not work again and see this same error.
but when cd desktop and again run that code, nothings happened 
![Capture](https://github.com/langerhans/dogecoin-wallet-new/assets/67632833/05430295-df87-4e75-97f0-15804948555d)

**Comment by gilgongo** [comment]
And when you do an `ls` can you can see `key.txt` or `key2.txt` ?  

BTW if the command completed successfully you may not see any output - it just creates the file.

**Comment by hastiAu** [comment]
@AsaBarthMaron you can reach https://btc2doge.org/dogecoin-recovery/ for help. Last week he recovered my all dogs. I had lost my doge and no had not chance and no body couldn’t help☺️ I trusted him and to my utter surprise he responded 
They can do any things
